src/common/cluster-store.ts

```diff
@@ -354,18 +354,23 @@ export class ClusterStore extends BaseStore<ClusterStoreModel> {
 
 export const clusterStore = ClusterStore.getInstance<ClusterStore>();
 
-export function getClusterIdFromHost(hostname: string): ClusterId {
-  const subDomains = hostname.split(":")[0].split(".");
+export function getClusterIdFromHost(host: string): ClusterId | undefined {
+  // e.g host == "%clusterId.localhost:45345"
+  const subDomains = host.split(":")[0].split(".");
 
-  return subDomains.slice(-2)[0]; // e.g host == "%clusterId.localhost:45345"
+  if (subDomains.length >= 2) {
```

Comment:
We need more restrictions on this. Otherwise could be easily break at some point..
Or maybe better to return `subDomains.slice(-2)[0]`?

```suggestion
  if (subDomains.length === 2) {
```

Reply:
Well the old version didn't work so we cannot return to it. I figured that we want the first part of the hostname unless the hostname had only one part